config/routes.rb

```diff
@@ -209,6 +209,10 @@
           resources :materials, path: 'files'
         end
       end
+
+      resources :course_users, only: [] do
+        resources :experience_points_records, only: [:new, :create, :index]
```

Comment:
I think you’re right about the EXP creation. The individual EXP awarding is not going to be useful. The main use case I know about is forum and tutorial participation, though I've not awarded either before.

Since this is the case, I think it might be good to redesign the underlying model as in #825 . What do you think?